blocks/editable/format-toolbar/style.scss

```diff
@@ -21,23 +20,41 @@
 	}
 }
 
+.blocks-format-toolbar__link-modal-line {
+	display: flex;
+	flex-direction: row;
+	flex-grow: 1;
+	flex-shrink: 1;
+	min-width: 0;
+	align-items: center;
+
+	.components-button {
+		flex-shrink: 0;
+	}
+}
+
 .blocks-format-toolbar__link-value {
-	padding: 10px;
+	margin: 10px;
 	flex-grow: 1;
+	flex-shrink: 1;
 	overflow: hidden;
 	position: relative;
+	white-space: nowrap;
+	min-width: 0;
 
 	&:after {
 		@include long-content-fade( $size: 40% );
 	}
 }
 
 .blocks-format-toolbar__link-settings {
+	padding: 7px 8px;
 	border-top: 1px solid $light-gray-500;
-	width: 100%;
-	padding: 10px 8px;
+	padding-top: 8px; // add 1px for the border
 
 	.blocks-base-control {
 		margin: 0;
+		flex-grow: 1;
+		flex-shrink: 1;
 	}
 }
```

Comment:
Are the `min-width: 0` necessary here?

Reply:
Strangely yes. Without the min-width it refuses to shrink.

Reply:
Nevermind, I thought you were talking about another part of the code. I have not needed to set min-width to zero on the link settings because it is smaller than the width of the link modal by default. It possibly doesn't need the flex-shrink.